src/scenes/Partnerships/PartnershipForm/PartnershipForm.tsx

```diff
@@ -0,0 +1,122 @@
+import { Box } from '@material-ui/core';
+import React from 'react';
+import { useFormState } from 'react-final-form';
+import {
+  displayPartnershipFundingType,
+  displayPartnershipStatus,
+  PartnershipFundingTypeList,
+  PartnershipStatuses,
+  PartnershipType,
+} from '../../../api';
+import {
+  DialogForm,
+  DialogFormProps,
+} from '../../../components/Dialog/DialogForm';
+import {
+  CheckboxesField,
+  CheckboxOption,
+  RadioField,
+  RadioOption,
+  SubmitError,
+} from '../../../components/form';
+import { OrganizationField } from '../../../components/form/Lookup';
+import { Nullable } from '../../../util';
+import { CreatePartnershipFormInput } from '../Create';
+import { EditPartnershipFormInput } from '../Edit';
+import { PartnershipFormFragment } from './PartnershipForm.generated';
+
+export type PartnershipFormProps<T> = DialogFormProps<T> & {
+  partnership?: PartnershipFormFragment;
+};
+
+export const hasManagingType = (types: Nullable<readonly PartnershipType[]>) =>
+  types?.includes('Managing') ?? false;
+
+export const PartnershipForm = <
+  T extends CreatePartnershipFormInput | EditPartnershipFormInput
+>({
+  partnership,
+  ...rest
+}: PartnershipFormProps<T>) => {
+  const radioOptions = PartnershipStatuses.map((status) => (
+    <RadioOption
+      key={status}
+      label={displayPartnershipStatus(status)}
+      value={status}
+    />
+  ));
+
+  return (
+    <DialogForm<T> {...rest}>
+      <SubmitError />
+      {!partnership && (
+        <OrganizationField
+          name="partnership.organizationId"
+          label="Organization"
+          required
+        />
+      )}
+      <CheckboxesField
+        name="partnership.types"
+        label="Types"
+        row
+        disabled={partnership ? !partnership.types.canEdit : false}
+      >
+        {([
+          'Managing',
+          'Funding',
+          'Impact',
+          'Technical',
+          'Resource',
+        ] as PartnershipType[]).map((type: PartnershipType) => (
+          <Box width="50%" key={type}>
```

Comment:
Can you replace `Box` with another styling solution?

Reply:
Haha, your nemesis!